Doc/c-api/object.rst

```diff
@@ -27,6 +27,24 @@ Object Protocol
    instead of the :func:`repr`.
 
 
+.. c:function:: int PyObject_HasAttrWithError(PyObject *o, const char *attr_name)
+
+   Returns ``1`` if *o* has the attribute *attr_name*, and ``0`` otherwise.
+   This is equivalent to the Python expression ``hasattr(o, attr_name)``.
+   On failure, return ``-1``.
```

Comment:
Given the relevance of error handling for these new functions, I think the docs should mention explicitly under which conditions users must expect exceptions.
```suggestion
   On failure, return ``-1`` with an exception set.
```